### PR TITLE
bing tts

### PR DESCRIPTION
**Pull request by JarbasAl**
add bing tts engine

### CODE REVIEW

**Comment by penrods** [comment]
Main question about this one is with regards to the bingtts requirements.  I started a similar discussion in #1262, so let's just about it there.

**Comment by forslund** [comment]
Tested this and ran into some trouble. I couldn't get the bingtts module to work at all with my access key.

Using the sample script from Microsoft, the access key works fine: (https://github.com/Azure-Samples/Cognitive-Speech-TTS/blob/master/Samples-Http/Python/TTSSample.py)

**Comment by forslund** [comment]
Figured it out, the gender value is case sensitive.

File: mycroft/tts/bing_tts.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+class BingTTS(TTS):
+    def __init__(self, lang, voice):
+        super(BingTTS, self).__init__(lang, voice, BingTTSValidator(self))
+        self.type = 'wav'
+        from bingtts import Translator
+        self.config = Configuration.get().get("tts", {}).get("bing", {})
+        api = self.config.get("api_key")
+        self.bing = Translator(api)
+        self.gender = self.config.get("gender", "male")
```
Comment (forslund) [inline mycroft/tts/bing_tts.py]: gender is case sensitive and need to be "Male"

**Comment by forslund** [comment]
Looks good, thank you! Merging!

**Pull request merged**